iceoryx2-ffi/ffi/src/api/service_builder_pub_sub.rs

```diff
@@ -442,6 +442,49 @@ pub unsafe extern "C" fn iox2_service_builder_pub_sub_set_max_subscribers(
     }
 }
 
+/// Enables/disables safe overflow for the service
+///
+/// # Arguments
+///
+/// * `service_builder_handle` - Must be a valid [`iox2_service_builder_pub_sub_ref_h`]
+///   obtained by [`iox2_service_builder_pub_sub`](crate::iox2_service_builder_pub_sub) and
+///   casted by [`iox2_cast_service_builder_pub_sub_ref_h`](crate::iox2_cast_service_builder_pub_sub_ref_h).
+/// * `value` - defines if safe overflow shall be enabled (true) or not (false)
+///
+/// # Safety
+///
+/// * `service_builder_handle` must be valid handles
+#[no_mangle]
+pub unsafe extern "C" fn iox2_service_builder_pub_sub_set_enable_safe_overflow(
+    service_builder_handle: iox2_service_builder_pub_sub_ref_h,
+    value: bool,
+) {
```

Comment:
Shall we remove the value? I guess it has a default value and since it is part of the builder it does not make sense to set it to the non-default value and then back. The Rust API does not need to be changed in this PR but maybe we can adjust the C and C++ API.

Depending on the default value, `activate_safe_overflow` or `deactivate_safe_overflow` would be an option.

Although, my favorite would be to use an enum like this
```rs
enum QueueSemantic/QueueMode {
    Fifo,
    RingBuffer/CircularBuffer,
}
```

Crossbeam, for example, uses the terminology ring-buffer for their `force_push` API